tests/test_env.py

```diff
@@ -269,21 +192,32 @@ def test_process_env_install_from_index(
     """
     package_name: str = "more-itertools"
     assert package_name not in env.process_env.get_installed_packages()
-    env.process_env.install_from_index([Requirement.parse(package_name + (f"=={version}" if version is not None else ""))])
+    env.process_env.install_for_config(
+        [Requirement.parse(package_name + (f"=={version}" if version is not None else ""))],
+        config=PipConfig(
+            use_system_config=True,  # we need an upstream for some packages
+        ),
+    )
     installed: Dict[str, version.Version] = env.process_env.get_installed_packages()
     assert package_name in installed
     if version is not None:
         assert installed[package_name] == version
 
 
 @pytest.mark.slowtest
+# TODO: depends on deciding exception structure
 def test_process_env_install_from_index_not_found(tmpvenv_active: Tuple[py.path.local, py.path.local]) -> None:
     """
     Attempt to install a package that does not exist from a pip index. Assert the appropriate error is raised.
     """
     with pytest.raises(env.PackageNotFound):
         # pass empty index list for security reasons (anyone could publish this package to PyPi)
```

Comment:
This comment used to refer to `index_urls=[]`, which is now gone. It should be moved to `use_system_config=False` (and updated) or removed.

Reply:
This test is a bit moot, as it will never even attempt to install, as there is not index set?

Reply:
Updated